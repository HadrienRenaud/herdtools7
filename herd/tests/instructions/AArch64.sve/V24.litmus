AArch64 V24
{
uint64_t 0:X0;
uint64_t 0:X1;
uint64_t 0:X2;
uint64_t 0:X3;
uint64_t 0:X4;
}
  P0        ;
CNTB X0,VL4 ;
INCB X0,VL4 ;
CNTW X1     ;
INCW X1     ;
CNTD X2,ALL,MUL 16 ;
INCD X2,ALL,MUL 16 ;
CNTW X3,MUL3 ;
INCW X3,MUL3 ;
CNTD X4,MUL4 ;
INCD X4,MUL4 ;
CNTW XZR     ;
forall
  0:X0=8
  /\ 0:X1=8 /\ 0:X2=64
  /\ 0:X3=6 /\ 0:X4=0
(* Notice: contents of registers 0:X1 to 0:X4 depend on where test runs *)
