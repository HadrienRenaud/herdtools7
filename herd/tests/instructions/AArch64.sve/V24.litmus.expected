Test V24 Required
States 1
0:X0=8; 0:X1=8; 0:X2=64; 0:X3=6; 0:X4=0;
Ok
Witnesses
Positive: 1 Negative: 0
Flag Scalable-Vector-Extensions-is-work-in-progress
Condition forall (0:X0=8 /\ 0:X1=8 /\ 0:X2=64 /\ 0:X3=6 /\ 0:X4=0)
Observation V24 Always 1 0
Hash=b8d825a2a1e147c6c00a1289b0020343

